pkg/asset/installconfig/openstack/validation/cloudinfo.go

```diff
@@ -133,22 +142,40 @@ func (ci *CloudInfo) getSubnet(subnetID string) (*subnets.Subnet, error) {
 	return subnet, nil
 }
 
-func (ci *CloudInfo) getFlavor(flavorName string) (*flavors.Flavor, error) {
+func (ci *CloudInfo) getFlavor(flavorName string) (Flavor, error) {
 	flavorID, err := flavorutils.IDFromName(ci.clients.computeClient, flavorName)
 	if err != nil {
 		var gerr *gophercloud.ErrResourceNotFound
 		if errors.As(err, &gerr) {
-			return nil, nil
+			return Flavor{}, nil
 		}
-		return nil, err
+		return Flavor{}, err
 	}
 
 	flavor, err := flavors.Get(ci.clients.computeClient, flavorID).Extract()
 	if err != nil {
-		return nil, err
+		return Flavor{}, err
+	}
+
+	var baremetal bool
+	{
+		const baremetalProperty = "baremetal"
+		var errNotFound *gophercloud.ErrResourceNotFound
+
+		m, err := flavors.GetExtraSpec(ci.clients.computeClient, flavorName, baremetalProperty).Extract()
```

Comment:
Shouldn't we pass the flavorID instead of flavorName?

Reply:
Absolutely, yes. Good catch, thanks